falcon_mcp/modules/sensor_usage.py

```diff
@@ -0,0 +1,99 @@
+"""
+Sensor Usage module for Falcon MCP Server
+
+This module provides tools for accessing CrowdStrike Falcon sensor usage data.
+"""
+
+from typing import Any, Dict, List, Optional
+
+from mcp.server import FastMCP
+from mcp.server.fastmcp.resources import TextResource
+from pydantic import AnyUrl, Field
+
+from falcon_mcp.common.errors import handle_api_response
+from falcon_mcp.common.logging import get_logger
+from falcon_mcp.common.utils import prepare_api_parameters
+from falcon_mcp.modules.base import BaseModule
+from falcon_mcp.resources.sensor_usage import SEARCH_SENSOR_USAGE_FQL_DOCUMENTATION
+
+logger = get_logger(__name__)
+
+
+class SensorUsageModule(BaseModule):
+    """Module for accessing CrowdStrike Falcon sensor usage data."""
+
+    def register_tools(self, server: FastMCP) -> None:
+        """Register tools with the MCP server.
+
+        Args:
+            server: MCP server instance
+        """
+        # Register tools
+        self._add_tool(
+            server=server,
+            method=self.search_sensor_usage,
+            name="search_sensor_usage",
+        )
+
+    def register_resources(self, server: FastMCP) -> None:
+        """Register resources with the MCP server.
+
+        Args:
+            server: MCP server instance
+        """
+        search_sensor_usage_fql_resource = TextResource(
+            uri=AnyUrl("falcon://sensor-usage/weekly/fql-guide"),
+            name="falcon_search_sensor_usage_fql_guide",
+            description="Contains the guide for the `filter` param of the `falcon_search_sensor_usage` tool.",
+            text=SEARCH_SENSOR_USAGE_FQL_DOCUMENTATION,
+        )
+
+        self._add_resource(
+            server,
+            search_sensor_usage_fql_resource,
+        )
+
+    def search_sensor_usage(
+        self,
+        filter: Optional[str] = Field(
```

Comment:
Based on #104 - let's remove Optional[]

```suggestion
        filter: str | None = Field(
```